### DISCUSSION TITLE
Dead Pinecil that will go into DFU and take a flash - can it be recovered?

### DISCUSSION THREAD

**Discussion by deftdawg** [Ralim/IronOS#988] [Q&A]
Pine Microsystems Inc.  `Support #4` directed me here after I opened a support ticket for a Pinecil that won't come on again after a single session of soldering.

According to them, because I am able to get it into DFU mode (holding down `-` while connecting to USB on Windows) and the Pinecil takes a flash from the firmware updating tool, the Pinecil is "working".   To me a soldering iron that doesn't solder isn't really "working"...

In any case, is there a way to get the Pinecil to come back to life or something I can do to debug it? 

Here's the output from flashing
```
Pinecil Firmware Updater v1.3
Looking for firmwares and latest version...
Downloading firmware...
Flashing...
dfu-util 0.10
Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2020 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/
Opening DFU capable USB device...
ID 28e9:0189
Run-time device DFU version 011a
Claiming USB DFU Interface...
Setting Alternate Setting #0 ...
Determining device status: state = dfuIDLE, status = 0
dfuIDLE, continuing
DFU mode device DFU version 011a
Device returned transfer size 2048
DfuSe interface name: "Internal Flash  "
Found GD32VF103, which reports a bad page size and count for its internal memory.
Fixed layout based on part number: page size 1024, count 128.
Downloading element to address = 0x08000000, size = 48800
Erase   	[                         ]   0%            0 bytes
Erase   	[=                        ]   4%         2048 bytes
Erase   	[==                       ]   8%         4096 bytes
Erase   	[===                      ]  12%         6144 bytes
Erase   	[====                     ]  16%         8192 bytes
Erase   	[=====                    ]  20%        10240 bytes
Erase   	[======                   ]  25%        12288 bytes
Erase   	[=======                  ]  29%        14336 bytes
Erase   	[========                 ]  33%        16384 bytes
Erase   	[=========                ]  37%        18432 bytes
Erase   	[==========               ]  41%        20480 bytes
Erase   	[===========              ]  46%        22528 bytes
Erase   	[============             ]  50%        24576 bytes
Erase   	[=============            ]  54%        26624 bytes
Erase   	[==============           ]  58%        28672 bytes
Erase   	[===============          ]  62%        30720 bytes
Erase   	[================         ]  67%        32768 bytes
Erase   	[=================        ]  71%        34816 bytes
Erase   	[==================       ]  75%        36864 bytes
Erase   	[===================      ]  79%        38912 bytes
Erase   	[====================     ]  83%        40960 bytes
Erase   	[======================   ]  88%        43008 bytes
Erase   	[=======================  ]  92%        45056 bytes
Erase   	[======================== ]  96%        47104 bytes
Erase   	[=========================] 100%        48800 bytes
Erase    done.
Download	[                         ]   0%            0 bytes
Download	[=                        ]   4%         2048 bytes
Download	[==                       ]   8%         4096 bytes
Download	[===                      ]  12%         6144 bytes
Download	[====                     ]  16%         8192 bytes
Download	[=====                    ]  20%        10240 bytes
Download	[======                   ]  25%        12288 bytes
Download	[=======                  ]  29%        14336 bytes
Download	[========                 ]  33%        16384 bytes
Download	[=========                ]  37%        18432 bytes
Download	[==========               ]  41%        20480 bytes
Download	[===========              ]  46%        22528 bytes
Download	[============             ]  50%        24576 bytes
Download	[=============            ]  54%        26624 bytes
Download	[==============           ]  58%        28672 bytes
Download	[===============          ]  62%        30720 bytes
Download	[================         ]  67%        32768 bytes
Download	[=================        ]  71%        34816 bytes
Download	[==================       ]  75%        36864 bytes
Download	[===================      ]  79%        38912 bytes
Download	[====================     ]  83%        40960 bytes
Download	[======================   ]  88%        43008 bytes
Download	[=======================  ]  92%        45056 bytes
Download	[======================== ]  96%        47104 bytes
Download	[=========================] 100%        48800 bytes
Download done.
File downloaded successfully
Your Pinecil was flashed successfully! You can disconnect it safely.
```

**Comment by HanaO00** [comment]
Hi, there are surely more competent people than me here, but I can possibly bring my stone to the building,
- Does the Pinecil heats up pressing "+" ? It may be a screen problem, if the Pine64 support indicates that the Pinecil is "working"
- Have you tried to flash another version of the firmware as a test ?

**Reply by deftdawg** [reply]
Your stone is appriecated 😄 

- sadly `+` doesn't do anything, I tried tapping it a number of times, holding it down, etc; tip stays cold.   I left it plugged to a power supply for a few days, when I picked it up, the screen woke, I moved to another room plugged in and then it was dead. 
 I'm think the screen should be physically ok, it wasn't dropped or jostled in anyway.
- For PSUs:
    -  First one I tried was a Dash charger, but it only puts out 5V - DC too low (Dash != QC); that was before my soldering session
    - Next I used an Asus Chromebox AC adapter -> 19V @ 3.42A which seemed fine for soldering until I disconnected it
    - After it died I also tried a HP AC adapter -> 19.5V @ 11.8A
    - Voltage of the two AC Adapters was validated by a multimeter
- I have noticed that to bottom area under the screen does get a little warm when the pinecil is connected to USB or a power supply, I wasn't paying attention to it before (was only gripping on the silicone sleeve) so that may or may not be something of interest
- Tried updating to 2.14.1 and 2.15, the flashing output is the same as above for both.  `Mass Erase` fails with `DFU-Util exited with error code 74!`, but I understand that is typical for many pinecils

**Comment by Ralim** [comment]
Hmm, I have no _special_ wisdom here really, if the MCU is flashing correctly and the screen isnt coming on that really only leaves the I2C bus. The parts on the I2C bus are: FUSB302 and the OLED itself.

I dont use the flasher tool so not 100% on the output, but could you try reading the firmware back out of the MCU just to confirm its a 1:1 copy (aka flashing worked correctly).

If the firmware has flashed correctly its possible its the OLED or the FUSB302.
If your comfortable it may be worthwhile opening it to see whats getting warm, as nothing should get overly warm from usb power.

**Reply by deftdawg** [reply]
So I tried to dump the data using dfu, don't think I'm doing it correcrly as it seems only 16k is being pulled from the device for both alt points.   So obviously gonna differ from the firmware .bin which is ~49K.


`dfu-util.exe -l`
```
dfu-util 0.10

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2020 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

Found DFU: [28e9:0189] ver=0100, devnum=28, cfg=1, intf=0, path="4-6", alt=1, name="@Option Bytes  /0x1FFFF800/01*016Be", serial="3TBJ"
Found DFU: [28e9:0189] ver=0100, devnum=28, cfg=1, intf=0, path="4-6", alt=0, name="@Internal Flash  /0x08000000/128*001Kg", serial="3TBJ"
```

`dfu-util.exe -a X -U read-from-device-X.bin` for alt 0 and alt 1
```
dfu-util.exe -a 1 -U read-from-device-1.bin
dfu-util 0.10

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2020 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

Opening DFU capable USB device...
ID 28e9:0189
Run-time device DFU version 011a
Claiming USB DFU Interface...
Setting Alternate Setting #1 ...
Determining device status: state = dfuIDLE, status = 0
dfuIDLE, continuing
DFU mode device DFU version 011a
Device returned transfer size 2048
Unbound upload not supported on DfuSe devices
Limiting default upload to 16384 bytes
Upload  [=========================] 100%        16384 bytes
Upload done.

C:\Users\User\Downloads\pinecil_firmware_updater_win64_1.3>dfu-util.exe -a 0 -U read-from-device-0.bin
dfu-util 0.10

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2020 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

Opening DFU capable USB device...
ID 28e9:0189
Run-time device DFU version 011a
Claiming USB DFU Interface...
Setting Alternate Setting #0 ...
Determining device status: state = dfuIDLE, status = 0
dfuIDLE, continuing
DFU mode device DFU version 011a
Device returned transfer size 2048
Unbound upload not supported on DfuSe devices
Limiting default upload to 16384 bytes
Upload  [=========================] 100%        16384 bytes
Upload done.
```

Agree, I think I'll try to take it apart tomorrow, I'm wondering if there's a fuse or something that fried inside.

**Reply by deftdawg** [reply]
`Found DFU: [28e9:0189] ver=0100, devnum=28, cfg=1, intf=0, path="4-6", alt=1, name="@Option Bytes  /0x1FFFF800/01*016Be", serial="3TBJ"`

Another stupid question, if dfu Alt `1` has saved options, is there a way to clear those options to factory defaults?

I didn't change any of the defaults (at least that I'm aware of) in my use of the device, but maybe there's a idle-timer/uptime counter or something that overflowed while it was sleeping for a few days or something dumb that's hose the config and prevented it from coming on...

**Reply by Ralim** [reply]
Ah if you have written to alt 1, you could have misconfigured any number of things.
Not entirely sure a non-tedius way to reset those without recalculating them (off memory they are _not_ 0xFF from factory).
I can try and dump some of my units tomorrow to have a look see.
Alt0 should let you dump at least 128K of flash, though I think you need to specify how much as dfu-util doesnt know exactly how much to dump by default

**Reply by Ralim** [reply]
All device settings are stored in the internal flash partition (alt 0) and if you do a mass erase you can reset them.
Alternatively if you flash a previous release and then come forward again generally that will causes a mismatch and it will reset them as well

**Reply by deftdawg** [reply]
> Ah if you have written to alt 1, you could have misconfigured any number of things.

Have not written to anything... According to the `dfu-util -help` upload is to download and download is to upload.

> I can try and dump some of my units tomorrow to have a look see.

That might be helpful, but I'm starting to suspect based on how hot this device gets, that it's fried.

> Alt0 should let you dump at least 128K of flash, though I think you need to specify how much as dfu-util doesnt know exactly how much to dump by default

```
$ ./dfu-util.exe -a 0 -Z 48800 -U read-firmware-v2.14.1.bin
dfu-util 0.10

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2020 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

Opening DFU capable USB device...
ID 28e9:0189
Run-time device DFU version 011a
Claiming USB DFU Interface...
Setting Alternate Setting #0 ...
Determining device status: state = dfuIDLE, status = 0
dfuIDLE, continuing
DFU mode device DFU version 011a
Device returned transfer size 2048
Limiting default upload to 16384 bytes
Upload  [=========================] 100%        16384 bytes
Upload done.
Unbound upload not supported on DfuSe devices
```

Doesn't seem to matter what I specify, it only wants to read 16K.   Maybe I've got the syntax wrong... 😞 


> All device settings are stored in the internal flash partition (alt 0) and if you do a mass erase you can reset them.

Mass Erase fails with `DFU-Util exited with error code 74!`

> Alternatively if you flash a previous release and then come forward again generally that will causes a mismatch and it will reset them as well

I've tried forward and back a few times.

**Reply by deftdawg** [reply]
So I took some thermal images of how it warms plugged into a USB, don't think its normal

|  Unplugged | ~30 sec |  ~5 min |
| ----------- | ----------- | ----------- |
| <img src="https://user-images.githubusercontent.com/4991612/120903899-f4964100-c616-11eb-9967-6e8be6798c39.jpg" width="300"> | <img src="https://user-images.githubusercontent.com/4991612/120903890-eba56f80-c616-11eb-8065-5e7abdf4891b.jpg" width="300"> | <img src="https://user-images.githubusercontent.com/4991612/120903898-f4964100-c616-11eb-93d1-521635a230cf.jpg" width="300"> |
|  Room Temperature is about 25C | hotspot |  Unit quite warm |

**Reply by tchebb** [reply]
FWIW, the correct command to readback the main flash with dfu-util is
```
dfu-util -d 28e9:0189 -a 0 -U main -s 0x08000000
```

If you're using dfu-util 0.10, the size should be autodetected, so no need for `-Z`. The `-d` is probably unnecessary if you only have the one device connected. The key part is `-s 0x08000000`, though, which tells it to read the main flash contents instead of the ROM bootloader, which is what your command would have read.

**Reply by deftdawg** [reply]
> FWIW, the correct command to readback the main flash with dfu-util is
> 
> ```
> dfu-util -d 28e9:0189 -a 0 -U main -s 0x08000000
> ```

Thanks @tchebb!  I was able to fetch the following binary (it was 128KB, but github requires a supported extension, so gzipped):
[main.gz](https://github.com/Ralim/IronOS/files/7782175/main.gz)

`binwalk` doesn't read anything, but running `strings` shows a few words to suggest it's not completely fried...
```
...
PolEng
pd_txTask
protRX
intTask
...
MCAUSE: 0x%lx
MEPC  : 0x%lx
MTVAL : 0x%lx
...
POWTask
GUITask
PIDTask
MOVTask
IDLE
...
```

**Reply by tchebb** [reply]
That dump exactly matches Pinecil_EN.bin from the [v2.14.1 release](https://github.com/Ralim/IronOS/releases/download/v2.14.1/Pinecil.zip). So I think your flashing worked correctly. I'm not a Pinecil expert, unfortunately (I don't even own one!), but I have worked with DFU mode on this chip before and was the one who added better support for it to dfu-util 0.10.

One slight oddity I noticed is that your flash has a bit of non-empty data past where the v2.14.1 image ends. That could be leftover data from a previous firmware version. It ought not to make any difference—well-written firmware will only ever look at initialized regions of flash—but in the interests of debugging you could try completely clearing the flash by running `dfu-util -d 28e9:0189 -a 0 -D -s 0x08000000:mass-erase:force` and reflashing. (If that gives an error, which it seems to on some version of the bootloader, you can get the equivalent effect by downloading an image with 128KiB of `0xff` bytes.) The DFU bootloader is in ROM, so the full erase has no risk of bricking the device further.

**Reply by tchebb** [reply]
If that doesn't change anything, seems like an IronOS bug that's showing up on your iron specifically for some reason. Or a hardware issue, I suppose.

**Reply by Ralim** [reply]
I would try a full flash erase to make sure you haven't gotten the stored settings / boot screen into a strange state.
Also just-to-be-sure try the latest master build which will cause a full settings reload
https://github.com/Ralim/IronOS/actions/runs/1625150942

Otherwise most likely thing is that the I2C bus is wedged and not operating correctly.
To solve that really requires removing the parts from the I2C line one at a time to find the cause or using a logic analyser to record the traffic to see what is going on

**Reply by deftdawg** [reply]
Ran `dfu-util -d 28e9:0189 -a 0 -s 0x08000000:mass-erase:force` (it didn't like the -D) and it did the following:

```
dfu-util 0.11

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2021 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

Opening DFU capable USB device...
Device ID 28e9:0189
Device DFU version 011a
Claiming USB DFU Interface...
Setting Alternate Interface #0 ...
Determining device status...
DFU state(2) = dfuIDLE, status(0) = No error condition is present
DFU mode device DFU version 011a
Device returned transfer size 2048
DfuSe interface name: "Internal Flash  "
Found GD32VF103, which reports a bad page size and count for its internal memory.
Fixed layout based on part number: page size 1024, count 128.
Performing mass erase, this can take a moment
dfu-util: Error during special command "MASS_ERASE" get_status
```

Next I did a `dfu-util -d 28e9:0189 -a 0 -U post-erase -s 0x08000000` and  a diff `diff main post-erase`, both came back the same...   So I guess I have to make a 128KB file of `0xFF`s and upload it to clear it out.

**Reply by deftdawg** [reply]
```sh
# OSX needs LANG=C for tr to produce ff
dd if=/dev/zero bs=1k count=64 | LANG=C tr '\000' '\377' > blank.bin

# -s 0x08000000:mass-erase:force -- always fails with `dfu-util: Error during special command "MASS_ERASE" get_status`
dfu-util -d 28e9:0189 -a 0 -D blank.bin -s 0x08000000
```

```sh
$ dfu-util -d 28e9:0189 -a 0 -U post-erase -s 0x08000000

dfu-util 0.11

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2021 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

Opening DFU capable USB device...
Device ID 28e9:0189
Device DFU version 011a
Claiming USB DFU Interface...
Setting Alternate Interface #0 ...
Determining device status...
DFU state(2) = dfuIDLE, status(0) = No error condition is present
DFU mode device DFU version 011a
Device returned transfer size 2048
DfuSe interface name: "Internal Flash  "
Found GD32VF103, which reports a bad page size and count for its internal memory.
Fixed layout based on part number: page size 1024, count 128.
Limiting upload to end of memory segment, 131072 bytes
Upload	[=========================] 100%       131072 bytes
Upload done.
```

Looks good:
```
$ diff main post-erase
Binary files main and post-erase differ
```

Doesn't mean anything to me, but looks like some `ff`s
```
$ hexdump post-erase
0000000 ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff
*
001fc00 24 00 40 01 96 00 05 00 00 1c 0a 00 05 00 d3 01
001fc10 a4 01 84 03 00 00 0a 00 01 00 01 00 00 00 00 00
001fc20 00 00 00 00 ff ff ff ff ff ff ff ff ff ff ff ff
001fc30 ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff
*
0020000
```

Push up the original firmware

```
dfu-util -d 28e9:0189 -a 0 -D Pinecil_EN.bin -s 0x08000000
```

```sh
dfu-util 0.11

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2021 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

dfu-util: Warning: Invalid DFU suffix signature
dfu-util: A valid DFU suffix will be required in a future dfu-util release
Opening DFU capable USB device...
Device ID 28e9:0189
Device DFU version 011a
Claiming USB DFU Interface...
Setting Alternate Interface #0 ...
Determining device status...
DFU state(2) = dfuIDLE, status(0) = No error condition is present
DFU mode device DFU version 011a
Device returned transfer size 2048
DfuSe interface name: "Internal Flash  "
Found GD32VF103, which reports a bad page size and count for its internal memory.
Fixed layout based on part number: page size 1024, count 128.
Downloading element to address = 0x08000000, size = 48800
Erase   	[=========================] 100%        48800 bytes
Erase    done.
Download	[=========================] 100%        48800 bytes
Download done.
File downloaded successfully
```

Hmmm... 
```
$ dfu-util -d 28e9:0189 -a 0 -U post-flash -s 0x08000000
dfu-util 0.11

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2021 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

Opening DFU capable USB device...
Device ID 28e9:0189
Device DFU version 011a
Claiming USB DFU Interface...
Setting Alternate Interface #0 ...
Determining device status...
DFU state(2) = dfuIDLE, status(0) = No error condition is present
DFU mode device DFU version 011a
Device returned transfer size 2048
DfuSe interface name: "Internal Flash  "
Found GD32VF103, which reports a bad page size and count for its internal memory.
Fixed layout based on part number: page size 1024, count 128.
Limiting upload to end of memory segment, 131072 bytes
Upload	[=========================] 100%       131072 bytes
Upload done.

$ diff Pinecil_EN.bin post-flash
Binary files Pinecil_EN.bin and post-flash differ
 ✖

$ diff post-flash ~/main
Binary files post-flash and /Users/deftdawg/main differ
 ✖
```

[post-flash.gz](https://github.com/Ralim/IronOS/files/7782502/post-flash.gz)

Disconnect USB-C and reconnect, still dead

Maybe I'll try with some newer firmware, since there's some other releases now.

**Reply by tchebb** [reply]
Use `cmp` instead of `diff` for the final check. `Pinecil_EN.bin` is much shorter than the full flash, so the files will differ if only because `post-flash` has a bunch of `0xff`s at the end. `cmp` will give a more detailed indication of what differs. You can also use `dhex` for a graphical comparison.

**Reply by deftdawg** [reply]
Oh, thanks for the tip on `cmp` and `dhex`, I already nuked the 2.14 firmware folder and tried with 2.16 RC2 ...   Also flashed okay, but device is still completely dead outside of dfu mode.   Pretty sure the hardware is toast, it's a shame because I only got 1 solder out of it, anyway, don't buy a pinecil before the next hardware revision drops for sure...  this rev is beautiful e-waste 😞 

```
$ dfu-util -d 28e9:0189 -a 0 -D Pinecil_EN.dfu
dfu-util 0.11

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2021 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

Multiple alternate interfaces for DfuSe file
Opening DFU capable USB device...
Device ID 28e9:0189
Device DFU version 011a
Claiming USB DFU Interface...
Setting Alternate Interface #0 ...
Determining device status...
DFU state(2) = dfuIDLE, status(0) = No error condition is present
DFU mode device DFU version 011a
Device returned transfer size 2048
DfuSe interface name: "Internal Flash  "
Found GD32VF103, which reports a bad page size and count for its internal memory.
Fixed layout based on part number: page size 1024, count 128.
File contains 1 DFU images
Parsing DFU image 1
Target name: ST...
Image for alternate setting 0, (1 elements, total size = 50832)
Setting Alternate Interface #0 ...
Parsing element 1, address = 0x08000000, size = 50824
Erase   	[=========================] 100%        50824 bytes
Erase    done.
Download	[=========================] 100%        50824 bytes
Download done.
Done parsing DfuSe file

$ dfu-util -l
dfu-util 0.11

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2021 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

Found Runtime: [0489:e07a] ver=0112, devnum=3, cfg=1, intf=3, path="20-2", alt=0, name="UNKNOWN", serial="ACD1B8E1AE18"
Found DFU: [28e9:0189] ver=0100, devnum=40, cfg=1, intf=0, path="20-3", alt=1, name="@Option Bytes  /0x1FFFF800/01*016Be", serial="3TBJ"
Found DFU: [28e9:0189] ver=0100, devnum=40, cfg=1, intf=0, path="20-3", alt=0, name="@Internal Flash  /0x08000000/128*001Kg", serial="3TBJ"
```

**Reply by Ralim** [reply]
The most likely thing here is just I2C bus issues, and that the MCU is operating just fine.

The Miniware irons are also plagued by OLED issues which is the most likely thing here. There are lots of posts around with people replacing them. It's just super low cost OLED's are not perfect.

If you have the ability, you could try an oled swap or removing the accelerometer or such if you wanted.

**Reply by deftdawg** [reply]
> The most likely thing here is just I2C bus issues, and that the MCU is operating just fine.
> 
> The Miniware irons are also plagued by OLED issues which is the most likely thing here. There are lots of posts around with people replacing them. It's just super low cost OLED's are not perfect.
> 
> If you have the ability, you could try an oled swap or removing the accelerometer or such if you wanted.

Maybe.... but I recall the OLED was `on` when I disconnected it after its only use... My guess whatever is causing this heat (see images below) is preventing power from being sent to the OLED or any other component.

> So I took some thermal images of how it warms plugged into a USB, don't think its normal
> 
> Unplugged	~30 sec	~5 min
> <img alt="" width="300" src="https://user-images.githubusercontent.com/4991612/120903899-f4964100-c616-11eb-9967-6e8be6798c39.jpg">	<img alt="" width="300" src="https://user-images.githubusercontent.com/4991612/120903890-eba56f80-c616-11eb-8065-5e7abdf4891b.jpg">	<img alt="" width="300" src="https://user-images.githubusercontent.com/4991612/120903898-f4964100-c616-11eb-93d1-521635a230cf.jpg">
> Room Temperature is about 25C	hotspot	Unit quite warm

**Reply by Ralim** [reply]
There has been more than one case of poor soldering on the oled, which can cause intermittent issues.
I would strongly suggest opening up the unit, touching up all OLED connections and testing then.


The unit will get warm normally so I wouldn't be panicking. Also what emissivity did you use? Was it calibrated?

**Reply by myself248** [reply]
I had identical symptoms and indeed it was an i2c fault. (Just got done troubleshooting it with @tchebb and @Ralim, thanks again!)  The iron would work after sitting for a while, but when I'd unplug/replug it wouldn't come up. And sometimes even if it started out working, it would randomly stop. But the whole time, DFU was fine. 

Turns out there was a bad solder join on the OLED GND pin. The i2c bus was being pulled high and low at the same time, which could explain your self-heating issue, and it's exactly the condition where DFU responds fine but the display never comes up. 
![IMG_0108](https://user-images.githubusercontent.com/4481636/147527381-b48174d8-0c69-4799-8846-58e8141c06e9.jpg)

Count down to the sixth pin from the top of that image. Soldered, soldered, soldered, soldered, soldered, marginal, soldered, soldered... It was a dicey proposition softening the foam adhesive to swing the OLED out of the way so I could inspect its solder joints, but as soon as I did, the problem was right there and the fix was easy enough. Have a look, and good luck!

**Reply by deftdawg** [reply]
> There has been more than one case of poor soldering on the oled, which can cause intermittent issues. I would strongly suggest opening up the unit, touching up all OLED connections and testing then.

I opened it up, honestly I wish I was as handy with a soldering iron as I am with a screwdriver...  Here are the joints for the OLED ribbon, they look "ok" to me, not sure that I could do better with my skill level.

<img alt="" width="400" src="https://user-images.githubusercontent.com/4991612/147531550-96fb02ef-ee0c-400b-9b42-1d265bdd8394.jpg"><img alt="" width="400" src="https://user-images.githubusercontent.com/4991612/147531547-88829d52-278b-4d15-a34e-d9d8918eb596.jpg">

> The unit will get warm normally so I wouldn't be panicking. Also what emissivity did you use? Was it calibrated?

The area getting hot is here, not sure which of the 3 components is the exact source (it's basically under the pine logo on the LED side):

![IMG_20211227_234057](https://user-images.githubusercontent.com/4991612/147531720-b59aa68e-0bd5-4128-a849-058b0597d00e.jpg)

**Reply by deftdawg** [reply]
> Count down to the sixth pin from the top of that image. Soldered, soldered, soldered, soldered, soldered, marginal, soldered, soldered... It was a dicey proposition softening the foam adhesive to swing the OLED out of the way so I could inspect its solder joints, but as soon as I did, the problem was right there and the fix was easy enough. Have a look, and good luck!

 Thanks, that's encouraging...  I used a flat head to stab the adhesive foam away from the board side, the OLED came loose rather suddenly at the end, but everything seems intact.

I'll have a go at it with my multimeter when I get back to the house tmr to see if there's continuity from ribbon to pad across each solder joint and if there's any bridging (continuity) with joints above and below each line.  Hopefully that will show me what to try to rework.

**Reply by myself248** [reply]
Wow, unless I'm mis-reading your photo, it looks like MOST of your joints are bad. I've annotated my image because I think it's clearer to start from, and because I was able to examine it from multiple angles to confirm my understanding:

![swoopylines](https://user-images.githubusercontent.com/4481636/147578657-6f045840-2068-4235-b4f1-aa4a7d449a54.jpg)

All the green lines show a continuous bead of solder from the board up to the OLED flex. The red line shows where there's a junction that didn't get made. It would sometimes make contact if the pressure and temperature(??) were just right, but sometimes not.

Looking at your image, it looks like _only the first and ninth_ have that nice continuous bead, all the rest appear to have a step profile rather than a swoosh profile, if that makes sense. 

I would pull the OLED completely off, clean the pads with braid, and reapply fresh flux and solder. I have some spare OLEDs here and I'd be happy to take some step-by-steps if that'd be helpful.

**Reply by deftdawg** [reply]
I got back to looking at it...  I'm afraid there's no way I'm going to be able to fix this (assuming this is the issue), it's simply at a scale that is too small for my eyes and my hands to work with.   I'm not even able to see the lines as I try to pin them with my multimeter, though I am able to get continuity across some while working blindly.

In any case, I think I'd need access to a microscope set-up (something like out of Louis Rossmann's workshop) and a way steadier hand than I have... 

Thanks all for the help.

**Reply by myself248** [reply]
Wow, you got those photos without a microscope? I'm impressed!

Yes, you'll definitely need optics for this. I'd try it with a swing-arm magnifier or a head-mounted visor thingy, both of which are considerably more affordable than the gold-standard binocular or Mantis scope. Does your local hackerspace/makerspace have any of the above? 

Or, since I think we can conclude with high confidence that it's an assembly defect, perhaps push back on Pine support and pursue a return/exchange. Best of luck.

**Reply by pwllheli** [reply]
I had the same problem. Thanks for the suggestion - it fixed it!
